c++/nda/mapped_functions.hxx

```diff
@@ -169,11 +171,10 @@ namespace nda {
   /// \return  A lazy nda::expr_call object (nda::Array) or the result of `std::exp` applied to the object (nda::Scalar).
   template <ArrayOrScalar A>
   auto exp(A &&a) requires(get_algebra<A> != 'M') {
-    return nda::map(
-       [](auto const &x) {
-         using std::exp;
-         return exp(x);
-       })(std::forward<A>(a));
+    auto scalar_lambda = [](auto const &x) { return std::exp(x); };
+    auto simd_lambda   = [](auto const &x) { return xsimd::exp(x); };
```

Comment:
in the end xsimd::exp should call std::exp for scalar under the hood. Shal we just use that in general to avoid the doubling of the code?